pallets/migrations/src/lib.rs

```diff
@@ -69,6 +70,13 @@ pub mod pallet {
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 		/// The list of migrations that will be performed
 		type MigrationsList: Get<Vec<Box<dyn Migration>>>;
+		/// The base call filter to be used in normal operating mode
+		/// (When we aren't in the middle of a migration)
+		type NormalCallFilter: Filter<Self::Call>;
+		/// The base call filter to be used when we are in the middle of migrations
+		/// This should be very restrictive. Probably not allowing anything except possibly
+		/// something like sudo or other emergency processes
+		type MigrationCallFilter: Filter<Self::Call>;
```

Comment:
Ah, nice idea on having two different filters (I had in mind trying to combine the logic into one single filter).

Reply:
We definitely want Sudo and perhaps Democracy to be able to do things like set storage in case of catastrophic failure...

Reply:
The difficulty is that democracy and sudo requires block production :/
Ideally, we should have a backup strategy `if current_slot > last_block.slot + (24 * 100)` like a revert of the code or state